docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
```

Comment:
TypeScript refers to these as "Type Aliases" in their docs, so please could we go for that name